### PR TITLE
Enhancement : Application cards and Accept/Reject button

### LINKED ISSUE #823: Enhancement : Make Application Cards Responsive and Accept/Reject Button Enhancement
### Issue Description
- The application cards for new users under Applications section is not properly shown on smaller screens (Margin on both side is not same). Also on larger screen, cards are displayed in two columns.
- The Accept / Reject Button in the cards do not show any visible change on hover or when the user clicks on it.

### Expected Behavior
- The application cards should be responsive on smaller screen (Margin should be same on both left/right side). Also cards should be displayed in one column with max width of 800px on larger screens.
- The Accept / Reject Button in the Application Cards should change color / show some effect on hover and when user clicks the button it should have some visible effect / action.

### Current Behavior
- Cards are shown in two columns and is not properly responsive on smaller screens.
- Accept / Reject button do not have any visible effect when the user hovers on it or when it is clicked.

### Screenshots

image

### Reproducibility

 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1. Go to https://dashboard.realdevsquad.com/applications/
2. You can see applications card shown two per row. Accept / Reject button do not have any hover effect and also it doesn't have any effect on click.

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [x] Medium
 - [ ] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by DeltaDynamo**
Date: 17 Sep 2024

Developer Name: [Anubhav Srivastava](https://github.com/DeltaDynamo)

---

## Issue Ticket Number

- Closes #823 

## Description

- Updated Reject Button element tag from `<p>` to `<button>`
- Updated CSS to make Application cards have 90% screen width (<500px screen size), 85% screen width (>500px screen size) with max card width of 800px and equal margins on both sides.
- Updated CSS to add effect on Click / Hover of Accept & Reject Buttons under Applications Card.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

#### Previous Behavior

https://github.com/user-attachments/assets/a7667efd-ce5b-48cc-be13-70479e7d5290

#### Post Change Behavior

https://github.com/user-attachments/assets/fef39706-d4c8-4bdb-a4bb-5c0a18c17cf3

#### Mobile View
description

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot 2024-09-17 204655

</details>

### CODE REVIEW

File: applications/style.css [hunk 1]
```diff
@@ -364,16 +367,61 @@ body {
...
+.application-details
+  .application-details-actions
+  .application-details-reject:hover,
+.application-details
+  .application-details-actions
+  .application-details-reject:active {
+  background: var(--red-color);
+  color: var(--white);
+  transition: transform 0.2s ease;
+}
```
Comment (Tejs1) [inline applications/style.css:385]: To simplify the CSS, use the :is() pseudo-class to group :hover and :active states. This avoids repeating the full selector:

```css

.application-details .application-details-actions .application-details-reject:is(:hover, :active) {
  background: var(--red-color);
  color: var(--white);
  transition: transform 0.2s ease;
}

```

**Review by Tejs1 — CHANGES_REQUESTED** [review]
use the :is() pseudo-class to group :hover and :active states.

File: applications/style.css (hunk 1 above)
Comment (DeltaDynamo) [reply applications/style.css:385]: Thank you for the wonderful suggestion! I will surely do that.

**Review by Tejs1 — APPROVED** [review]

**Review by Tejs1 — APPROVED** [review]

File: applications/index.html [hunk 2]
```diff
@@ -78,7 +78,7 @@ <h2>Status</h2>
         <div class="application-details-main"></div>
         <div class="application-details-actions">
           <button class="application-details-accept">Accept</button>
-          <p class="application-details-reject">Reject</p>
+          <button class="application-details-reject">Reject</button>
```
Comment (Achintya-Chatterjee) [inline applications/index.html:81]: Also, you can consider keyboard focus states (`:focus-visible`) for both buttons so users relying on keyboards can easily navigate.

File: applications/style.css [hunk 3]
```diff
@@ -364,16 +367,55 @@ body {
...
+  .application-details-reject:is(:hover, :active) {
+  background: var(--red-color);
+  color: var(--white);
+  transition: transform 0.2s ease;
+}
+
+.application-details
+  .application-details-actions
+  .application-details-reject:hover {
+  transform: scale(1.05);
```
Comment (Achintya-Chatterjee) [inline applications/style.css:387]: Including `transform` for the scaling effect is good, but keep in mind the performance implications of CSS animations on low-end devices. To optimize performance, you could use `will-change: transform`.

File: applications/index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (DeltaDynamo) [reply applications/index.html]: I have added `aria-labels` for the buttons.

Comment (DeltaDynamo) [reply applications/index.html]: Thanks for pointing out! updated.

File: applications/style.css (hunk 3 above)
Comment (DeltaDynamo) [reply applications/style.css:388]: Thanks for the feedback! I've added will-change: transform to optimize the scaling effect for better performance on low-end devices.

**Review by Achintya-Chatterjee — APPROVED** [review]
LGTM

**Review by lakshayman — APPROVED** [review]

**Pull request merged**